deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/ConfigHelper.java

```diff
@@ -136,14 +136,15 @@ public static TeletraanServiceContext setupContext(TeletraanServiceConfiguration
         context.setMailManager(configuration.getEmailFactory().createMailManager());
         context.setHostGroupDAO(configuration.getHostGroupFactory().createHostGroupDAO());
 
+        String defaultScmTypeName = configuration.getDefaultScmTypeName();
         List<SourceControlFactory> sourceControlConfigs = configuration.getSourceControlConfigs();
         HashMap<String, SourceControlManager> managers = new HashMap<String, SourceControlManager>();
         for(SourceControlFactory scf : sourceControlConfigs) {
             SourceControlManager scm = scf.create();
-            String type = scm.getType();
+            String type = scm.getTypeName();
             managers.put(type, scm);
```

Comment:
I think this becomes a bug based on your implementation. `scm.getTypeName()` returns 'Github' and then `buildBean.getScm()` returns 'GITHUB`, `hasSCMType()` will always return false.

Reply:
`StringUtils.capitalize(arg)` returns a String with only the first letter capitalized so `buildBean.getScm()` returns 'Github'

Reply:
I see. I thought it did all cap. Then could you use either all cap or all lower case? Having only the first letter capitalized is not a very good practice.

Reply:
I see what you mean. We would be modifying each scm from the Json to be all caps. I will update the typeName in teletraanconfigs to reflect that as well so that when the scm is being checked against the valid scm list it doesn't result in an error

Reply:
I'm getting a few test fails because when I call `toUpperCase()` on `scm` within the `genSetClause()` method in those tests it returns a nulPointerException because scm is not populated in those tests with`toUpperCase()` still being called on the `scm` variable. I also tested adding the scm value to the broken tests and that seemed to break other tests related to other beans like `TagBean`

Reply:
I committed the updates along with some unit test updates that were needed